### DISCUSSION TITLE
Are Channels the recommended way to add different modes to an instrument?

### DISCUSSION THREAD

**Discussion by mmerlo** [pymeasure/pymeasure#1158] [Q&A]
What is the recommended way to add different modes to an instrument? 

For example, let's say I have a 1-channel oscilloscope with two modes. Mode 1 is normal acquisition mode and Mode 2 is FRA (frequency response analysis). These two modes have different parameters.

Are "Channels" the best way to implement the different modes of this instrument? If not, what is the recommended way to add modes to an instrument?

Thanks.

**Comment by BenediktBurger** [comment]
What does change with these modes?
Are they mutually exclusive?

Currently, pymeasure does not have a system to do things dependent on the state ("mode") of a device known to the driver.

As we sometimes use channels to group properties/methods, they could be used as they offer the same name dependent on the "mode" and you can tell people to use one mode or the other.
An implementation idea: The channels could have an "activate" method.
